File: 27.removeelement/removeelement.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        fastIndex = 0
        slowIndex = 0
        for fastIndex in range(0,len(nums)):
            if nums[fastIndex] != val:
                nums[slowIndex] = nums[fastIndex]
                slowIndex = slowIndex + 1
        
        return slowIndex

File: arrayproblems/27.removeelement/removeelement.java
// 双指针
class Solution {
    public int removeElement(int[] nums, int val) {
        int fastIndex;
        int slowIndex = 0;
        for(fastIndex = 0; fastIndex<nums.length;fastIndex++){
            if(nums[fastIndex] != val){
                nums[slowIndex]= nums[fastIndex];
                slowIndex++;
            }
            
        }
        return slowIndex;
    }
}